iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -1071,13 +1078,11 @@ void PortManager::addPublisherToServiceRegistry(const capro::ServiceDescription&
         IOX_LOG(WARN) << "Could not add publisher with service description '" << service << "' to service registry!";
         errorHandler(PoshError::POSH__PORT_MANAGER_COULD_NOT_ADD_SERVICE_TO_REGISTRY, ErrorLevel::MODERATE);
     });
-    publishServiceRegistry();
 }
 
 void PortManager::removePublisherFromServiceRegistry(const capro::ServiceDescription& service) noexcept
 {
     m_serviceRegistry.removePublisher(service);
-    publishServiceRegistry();
```

Comment:
Oh boy, I remember the bikeshedding regarding the right (i.e. fastest) data structure for the `ServiceRegistry` well enough but it seems after the implementation nobody bothered to look at the slow tests with flamegraph to discover that there is something an order of magnitude slower than the most naive data structure one can use. This is a prime example why one should measure first and then optimize the bottleneck.